src/bloom/scalable-bloom-filter.ts

```diff
@@ -0,0 +1,218 @@
+   */
+  @Field()
+  public _ratio: number
+
+  /**
+   * Internal Partition Bloom Filters
+   */
+  @Field(
+    (filters: PartitionBloomFilter[]) =>
+      filters.map(filter => filter.saveAsJSON()), // eslint-disable-line @typescript-eslint/no-unsafe-return
+    (array: []) =>
+      array.map(
+        data => PartitionBloomFilter.fromJSON(data) as PartitionBloomFilter
+      )
+  )
+  public _filters: PartitionBloomFilter[] = []
+
+  constructor(
+    @Parameter('_initial_size')
+    _initial_size = 8,
+    @Parameter('_error_rate')
+    _error_rate = 0.01,
+    @Parameter('_ratio')
+    _ratio = 0.5
+  ) {
+    super()
+    this._initial_size = _initial_size
+    this._error_rate = _error_rate
+    this._ratio = _ratio
+    this._filters.push(
+      PartitionBloomFilter.create(
+        this._initial_size,
+        this._error_rate,
+        this._ratio
+      )
+    )
+    this._filters[this._filters.length - 1].seed = this.seed
+  }
+
+  /**
+   * @override
+   * Return the current seed.
+   * For obscure reason we must code this function...
+   */
+  public get seed() {
+    return this._seed
+  }
+
+  /**
+   * @override
+   * Set the seed for this structure. If you override the seed after adding data,
+   * all the filters will be updated and you may get wrong indexes for already indexed data
+   * due to the seed change. So only change it once before adding data.
+   * @param  seed the new seed that will be used in this structure
+   */
+  public set seed(seed: number) {
+    this._seed = seed
+    this._rng = seedrandom(`${this._seed}`) as prng
+    this._filters.forEach((filter: PartitionBloomFilter) => {
+      filter.seed = this.seed
+    })
+  }
+
+  /**
+   * Add a new element to the filter
+   * @param element
+   */
+  public add(element: HashableInput) {
+    // determine if we need to create a new filter
+    const currentFilter = this._filters[this._filters.length - 1]
+    if (currentFilter._currentload() > currentFilter._loadFactor) {
+      // create a new filter
+      const newSize =
+        this._initial_size *
+        Math.pow(ScalableBloomFilter._s, this._filters.length + 1) *
+        Math.LN2
+      const newErrorRate =
+        this._error_rate * Math.pow(this._ratio, this._filters.length)
+      this._filters.push(
+        PartitionBloomFilter.create(newSize, newErrorRate, this._ratio)
+      )
+      this._filters[this._filters.length - 1].seed = this.seed
+    }
+    // get the newly created filter
+    this._filters[this._filters.length - 1].add(element)
+  }
+
+  /**
+   * Return True if the element has been found, false otherwise.
+   * Check until we found the value in a filter otherwise stop on the first value found.
+   * @param element
+   * @returns
+   */
+  public has(element: HashableInput) {
+    let i = 0
+    let found = false
+    while (!found && i < this._filters.length) {
+      found = this._filters[i].has(element)
+      i++
+    }
+    return found
```

Comment:
I think the following is more compact and strictly equivalent
```suggestion
    return this._filters.some((filter) => filter.has(element))
```

Reply:
`.some` iterates only on a fixed array and do not iterate on inserted elements after the iteration begins while the `while` loop does allow it. For mono-threaded environment your suggestion is ok. +1